File: PitRadio.Api/Data/Model/Album.cs
using System;
using System.Collections.Generic;

namespace PitRadio.Api.Data.Model
{
    public class Album
    {
        public Album(string uuid, string title, ushort year, string artist, string folder, IEnumerable<Song> songs)
        {
            UUID = uuid;
            Title = title;
            Year = year;
            Artist = artist;
            Folder = folder;
            Songs = songs;
        }

        public string UUID { get; }
        public string Title { get; }
        public ushort Year { get; }
        public string Artist { get; }
        public string Folder { get; }
        public IEnumerable<Song> Songs { get; }
    }
}

File: frontend/lib/album.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pitradio/song.dart';

part 'album.g.dart';

@JsonSerializable()
class Album {
  Album({
    required this.artist,
    required this.title,
    required this.uuid,
    required this.songs,
  });

  final String uuid;
  final String title;
  final String artist;
  final List<Song> songs;

  // final String author;
  // final String thumbnailUrl;

  factory Album.fromJson(Map<String, dynamic> json) => _$AlbumFromJson(json);
  Map<String, dynamic> toJson() => _$AlbumToJson(this);
}
